app/api/agent-templates/route.ts

```diff
@@ -124,5 +124,66 @@ export async function DELETE(request: Request) {
   }
 }
 
+export async function PATCH(request: Request) {
+  try {
+    const body = await request.json();
+    const {
+      id,
+      userId,
+      title,
+      description,
+      prompt,
+      tags,
+      isPrivate,
+    }: {
+      id: string;
+      userId: string;
+      title?: string;
+      description?: string;
+      prompt?: string;
+      tags?: string[];
+      isPrivate?: boolean;
+    } = body;
+
+    if (!id || !userId) {
+      return NextResponse.json({ error: "Missing id or userId" }, { status: 400 });
+    }
+
+    // Verify ownership
+    const { data: template, error: fetchError } = await supabase
+      .from("agent_templates")
+      .select("id, creator")
+      .eq("id", id)
+      .single();
```

Comment:
Single Responsibility Principle
- actual: Supabase lib defined within API route
- required: Supabase lib defined in `/lib/supabase` as a standalone file.